Detectors/FIT/common/simulation/include/FITSimulation/Digitizer.h

```diff
@@ -65,18 +67,33 @@ class Digitizer
   Int_t mEventID;
   Int_t mSrcID;        // signal, background or QED
   Double_t mEventTime; // timestamp
+  int mNoisePeriod;    //low frequency noise period
+  int mBinshift;           // number of bin to shift positive part of CFD signal
 
   DigitizationParameters parameters;
 
   o2::dataformats::MCTruthContainer<o2::t0::MCLabel>* mMCLabels = nullptr;
 
   static constexpr Float_t C_side_cable_cmps = 2.877; //ns
   static constexpr Float_t A_side_cable_cmps = 11.08; //ns
+
+  TH1F* mHist;      // ("time_histogram", "", 1000, -0.5 * signal_width, 0.5 * signal_width);
+  TH1F* mHistsum;   //("time_sum", "", 1000, -0.5 * signal_width, 0.5 * signal_width);
+  TH1F* mHistshift; //("time_shift", "", 1000, -0.5 * signal_width, 0.5 * signal_width);
+
   //static constexpr Float_t signal_width = 5.;         // time gate for signal, ns
 
   //static std::vector<double> aggregate_channels(const std::vector<o2::fit::HitType>& hits, DigitizationParameters const& parameters);
 
   ClassDefNV(Digitizer, 1);
+};
+ inline double sinc(const double x)
+{
+   return (std::abs(x)<1e-12) ? 1 : std::sin(x)/x; 
+}
+inline double signalForm_i(double x)
+{
+  return -(exp(-0.83344945 * x) - exp(-0.45458 * x)) * (x >= 0) / 7.8446501; // Maximum should be 7.0/250 mV
```

Comment:
@AllaMaevskaya what I meant is that here you do relatively heavy calculations of exp then multiply it by 0 when ``x<0`` (as far as I can see, 50% of cases). Therefore, it will be faster to do
````
return x>0 ? -(exp(-0.83344945 * x) - exp(-0.45458 * x))/7.8446501 : 0.;
````

Reply:
OK, done.